src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1707,6 +1671,43 @@ private FeedbackSessionResultsBundle getFeedbackSessionResultsForUserInSectionBy
                         visibilityTable, responseStatus, roster, responseComments);
     }
 
+    private Map<String, String> initializeParamsWithSelectedSectionDetail(String sectionDetail) {
+
+        Map<String, String> params = new HashMap<>();
+        // this happens when 'all' section is selected
+        if (sectionDetail == null) {
+            // default for 'all' section selected
+            params.put(PARAM_IN_SECTION, "true");
+            params.put(PARAM_FROM_SECTION, "false");
+            params.put(PARAM_TO_SECTION, "false");
+            return params;
+        }
+        // either giver or recipient needs to be in the selected section
+        if (sectionDetail.contains("either")) {
```

Comment:
Should we define `sectionDetails` as an enum?

Reply:
Right now, sectionDetail will be a long text from the selected option using `params.get(PARAM_SECTION_DETAIL)`. By using enum, I can map the long text to keyword based but it will be additional process. 
So I am not sure if it is necessary?

Reply:
I kind of have some feelings:

- It would be hard to maintain and understand: It is hard to infer what the `sectionDetails` is as it is just a `String` (and you also allow the possibility of `null`). We can mitigate this by writing javadoc, but I still think `enum` could be a better option as developers can just navigate to the enum.

- It would be hard to add options in the futher: What if you have a option that `contains` words of another options. Are you going to have nested `if`?

- By right, `sectionDetails` is not a variable but rather a option. It should associate with a fixed set of value.

Reply:
Hmm I see. In terms of maintainability, enum seems better. Working on it!